src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -849,6 +851,22 @@ void BlockLocalPositionEstimator::predict()
 	_P += (_A * _P + _P * _A.transpose() +
 	       _B * _R * _B.transpose() +
 	       _Q) * getDt();
+
+	// condition P
+	for (int i=0; i<n_x; i++) {
+		for (int j=0; j<i; j++) {
+			if (_P(i, j) < 0) {
```

Comment:
Here we check if all elements on the matrix are positive, but, isn't only the diagonal meant to be positive?